From 3506ee9262d0a96a42b14fc99011e9349b1a121a Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Wed, 7 Feb 2018 16:18:33 +0800
Subject: [PATCH] Revert "gpio: rockchip: Make the gpio base added at gpiolib"

This reverts commit 545abf2ec49211df4d15e9ddd7bc128334fc18ad.

Change-Id: Icb7a08f12e85a8757154356a6671b5bca88eb79a
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 drivers/gpio/gpiolib.c             | 11 ++++-------
 drivers/pinctrl/pinctrl-rockchip.c |  2 +-
 2 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/drivers/gpio/gpiolib.c b/drivers/gpio/gpiolib.c
index 3c5ea1ce2f9e..06d345b087f8 100644
--- a/drivers/gpio/gpiolib.c
+++ b/drivers/gpio/gpiolib.c
@@ -126,7 +126,7 @@ EXPORT_SYMBOL_GPL(gpiod_to_chip);
 static int gpiochip_find_base(int ngpio)
 {
 	struct gpio_chip *chip;
-	int base = ARCH_NR_GPIOS - ngpio + ARCH_GPIO_BASE;
+	int base = ARCH_NR_GPIOS - ngpio;
 
 	list_for_each_entry_reverse(chip, &gpio_chips, list) {
 		/* found a free space? */
@@ -301,21 +301,18 @@ int gpiochip_add(struct gpio_chip *chip)
 	unsigned long	flags;
 	int		status = 0;
 	unsigned	id;
-	int		base;
+	int		base = chip->base;
 	struct gpio_desc *descs;
 
-	chip->base += ARCH_GPIO_BASE;
-	base = chip->base;
-
 	descs = kcalloc(chip->ngpio, sizeof(descs[0]), GFP_KERNEL);
 	if (!descs)
 		return -ENOMEM;
 
 	spin_lock_irqsave(&gpio_lock, flags);
 
-	if (base < ARCH_GPIO_BASE) {
+	if (base < 0) {
 		base = gpiochip_find_base(chip->ngpio);
-		if (base < ARCH_GPIO_BASE) {
+		if (base < 0) {
 			status = base;
 			spin_unlock_irqrestore(&gpio_lock, flags);
 			goto err_free_descs;
diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index 0073fa7e7c20..640bca07428d 100644
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -3183,7 +3183,7 @@ static int rockchip_gpiolib_register(struct platform_device *pdev,
 		bank->gpio_chip = rockchip_gpiolib_chip;
 
 		gc = &bank->gpio_chip;
-		gc->base = bank->pin_base;
+		gc->base = ARCH_GPIO_BASE + bank->pin_base;
 		gc->ngpio = bank->nr_pins;
 		gc->dev = &pdev->dev;
 		gc->of_node = bank->of_node;
-- 
2.35.3

